### DISCUSSION TITLE
How to en/decode time.Time by `json`.

### DISCUSSION THREAD

**Discussion by changrui** [vlang/v#11793] [Ideas]
```
fn encode_date() {
    t := time.new_time(year: 2020,   month: 8,  day: 13, )    
    s := json.encode(t)
    println('Encode Date: ${s}.')   //Got: {"year":2020,"month":8,"day":13,"hour":0,"minute":0,"second":0,"microsecond":0,"unix":1597276800}
}
 
fn decode_date() {
    s := '2020-9-13'
    d := json.decode(time.Time, s) or {
        println('Decode error: ${err}.')
        return 
    }    
    println('Decode: ${d}.')        //Got: 0000-00-00 00:00:00
}
```
I need to decode `2020-9-13` to `time.Time` struct, who would like to help me? Thanks a lot.

**Comment by changrui** [comment]
decode `2020-9-13` to `time.Time` manually as fellow:
```
    e := s.split('-')
    f :=time.Time{                    // or time.new_time(...)
        year:    e[0].int()
        month:   e[1].int()
        day:     e[2].int()
    }
``` 
It is so ugly!  omg.

**Comment by medvednikov** [comment]
time.Time is now supported natively by json, but only if it's used as a struct field.

**Comment by changrui** [comment]
> time.Time is now supported natively by json, but only if it's used as a struct field.

But I had test it at `devbits.app/play`, and got result as above, why?

**Comment by JalonSolov** [comment]
As he said, "but only if it's used as a struct field." - it won't work as a regular variable (for now), only if it is a field in a struct.

**Comment by changrui** [comment]
> As he said, "but only if it's used as a struct field." - it won't work as a regular variable (for now), only if it is a field in a struct.

oh, No. Got:  `Decode: Date{Time: 1969-12-31 00:00:00}.`
my code here:
```
import time
import json

fn decode_date() {
    s := '2020-9-13'
    d := json.decode(Date, s) or {
        println('Decode error: ${err}.')
        return 
    }    
    println('Decode: ${d}.')        
}

fn main() {
    decode_date()
}

struct Date {
	time.Time
}
```

**Comment by JalonSolov** [comment]
`1969-12-31 00:00:00` is the `zero` value for time.

**Comment by changrui** [comment]
oh?  It should be `1970-1-1 00:00:00`?!  But my time is `2020-9-13`, why?

**Comment by changrui** [comment]
oh, God! I change a bit, `s := 'date: 2020-9-13'`, got: `*Decode error*: date: 2020-9-13`, so strange!!!
